source/persistence/MessageController.cpp

```diff
@@ -170,6 +172,34 @@ bool MessageController::addMessage(const QString &id, const QString &jid, const
     }
     else
     {
```

Comment:
Else will also be processed for MaM messages. It would be fine to have a more explicit update for httpUpload stuff only. I dont have a good solution in mind now. Let me think about that issue...